libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -207,10 +207,11 @@ pub struct NotificationHandler {
 }
 
 impl NotificationHandler {
-    pub fn listen(&self) -> Result<CmNotification, ConnectionManagerError> {
-        match self.recv.recv() {
-            Ok(notifications) => Ok(notifications),
-            Err(_) => Err(ConnectionManagerError::SendMessageError(
+    pub fn try_next(&self) -> Result<Option<CmNotification>, ConnectionManagerError> {
+        match self.recv.try_recv() {
```

Comment:
What's the justification for this? There's not information in the commit message explaining why this is the case.

It seems like it may just introduce busy-wait situation for the user. Shouldn't there still be the blocking version of this method?

Reply:
Updated commit message

Reply:
This is now hidden behind a feature